web/pdf_find_controller.js

```diff
@@ -799,34 +820,22 @@ class PDFFindController {
         })
         .join("|");
     }
+    if (!query) {
+      return undefined;
+    }
 
+    const { caseSensitive, entireWord } = this.#state;
     const flags = `g${isUnicode ? "u" : ""}${caseSensitive ? "" : "i"}`;
-    query = query ? new RegExp(query, flags) : null;
-
-    this.#calculateRegExpMatch(query, entireWord, pageIndex, pageContent);
+    query = new RegExp(query, flags);
 
-    // When `highlightAll` is set, ensure that the matches on previously
-    // rendered (and still active) pages are correctly highlighted.
-    if (this.#state.highlightAll) {
-      this.#updatePage(pageIndex);
-    }
-    if (this._resumePageIdx === pageIndex) {
-      this._resumePageIdx = null;
-      this.#nextPageMatch();
-    }
-
-    // Update the match count.
-    const pageMatchesCount = this._pageMatches[pageIndex].length;
-    this._matchesCountTotal += pageMatchesCount;
-    if (this.#updateMatchesCountOnProgress) {
-      if (pageMatchesCount > 0) {
-        this.#updateUIResultsCount();
+    const matches = [];
+    for (const { index, 0: match } of pageContent.matchAll(query)) {
```

Comment:
Sorry, but this looks like a completely unrelated change that we shouldn't "sneak" into *this* patch.